### PR TITLE
Initial cut at add_gif.sh script

### PR DESCRIPTION
**Pull request by daynemay**
Allows someone (maybe with the _exception_ of @npars) to create a PR for a new +1 GIF with a single command nominating only a proposed name for the GIF, and its URL, e.g.

`./add_gif.sh test https://media.giphy.com/media/Zp3dDTwtkKKU8/giphy.gif`

### CODE REVIEW

**Comment by dpatterson-aio** [comment]
amazing!

File: add_gif.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Comment (npars) [inline add_gif.sh]: Standard is `#!/usr/bin/env bash`

File: add_gif.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+# Usage: ./add_gif.sh name http://url_of_the_gif
+NAME=$1
+URL=$2
+
+# Usage 
+if [ "$NAME" = "--help" ]; then
```
Comment (npars) [inline add_gif.sh]: When using bash, you should prefer the bash test built-in `[[`. How about we print help if the arg count is off:
`[[ "$#" -ne 3 ]]`

**Review by npars — COMMENTED** [review]
I'm wondering if this should be split into two scripts, one that does the retrieval/processing, and one that does git/github related things

**Comment by daynemay** [comment]
Split: seems reasonable. All the Git/Github relies on a particular flow, so (as much as I would love to be able to bug you with a single command), maybe I'll cut this back to just `wget` the file and run `process_gifs.sh` on it, and leave all the branching, committing, PRing up to the user.

File: add_gif.sh (hunk 2 above)
Comment (daynemay) [reply add_gif.sh]: (but 2, since `$#` is a count of the arguments, i.e. excludes the command itself)

File: add_gif.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+fi
+
+# Ensure availability of wget
+if ! type wget > /dev/null 2> /dev/null; then
+    echo "Script requires wget"
+    exit 2
+fi
+
+# Ensure name is not already in use
+if [[ -f ./_orig/$NAME.gif ]]; then
```
Comment (npars) [inline add_gif.sh]: This is going to explode if $NAME has a space in it, but in that case I don't know if the output would work anyway? Maybe we should prevent names with spaces and other strange characters?

Comment (daynemay) [reply add_gif.sh]: Latest commit restricts to alphanumerics, hyphen, underscore.

**Pull request merged**